table/block_based/block_based_table_builder.cc

```diff
@@ -505,9 +505,12 @@ struct BlockBasedTableBuilder::Rep {
           use_delta_encoding_for_index_values, p_index_builder_));
     }
 
-    for (auto& collector_factories : *tbo.int_tbl_prop_collector_factories) {
+    const auto& factory_range = tbo.int_tbl_prop_collector_factories;
+    for (auto it = factory_range.first; it != factory_range.second; ++it) {
+      assert(*it);
+
       table_properties_collectors.emplace_back(
-          collector_factories->CreateIntTblPropCollector(column_family_id));
+          (*it)->CreateIntTblPropCollector(column_family_id));
```

Comment:
Would it be overdesign to have a factory_range class that has an `begin/end`? which basically has the factories and the range in the same place.

Reply:
Do you mean putting the vector of factories and the range (i.e. the two iterators) in the same class? That wouldn't work since the vector has unique ownership of the factories; what we need instead is a way to refer to either the entire range or just a subrange of the same vector of factories.